libparsec/crates/crypto/src/sodiumoxide/sequester.rs

```diff
@@ -399,9 +400,33 @@ impl SequesterVerifyKeyDer {
         match verifier.verify(signature) {
             Ok(signature_is_valid) => match signature_is_valid {
                 true => Ok(contents.to_vec()),
-                false => Err(CryptoError::Signature),
+                false => Err(CryptoError::SignatureVerification),
             },
             Err(_) => Err(CryptoError::SignatureVerification),
         }
     }
+
+    pub fn unsecure_unwrap(signed: &[u8]) -> Result<(&[u8], &[u8]), CryptoError> {
+        if signed.len() <= Self::ALGORITHM.len() {
+            return Err(CryptoError::Signature);
+        }
+
+        let (_algorithm, others) = signed.split_at(Self::ALGORITHM.len());
+
+        let signature_size = match others.first() {
+            // Signature: algo '@' rsa_sign_size ':' rsa_signature
+            Some(b'@') if others.len() >= 3 => {
+                1 + 2 + u16::from_le_bytes([others[1], others[2]]) as usize
+            }
```

Comment:
We need to check if `others[3] == ':'`

Reply:
`unsecure_unwrap` does not need to know if signature is good.